tooling/api/src/window.ts

```diff
@@ -896,9 +1022,75 @@ class WindowManager extends WebviewWindowHandle {
     return invokeTauriCommand({
       __tauriModule: 'Window',
       message: {
-        cmd: 'startDragging'
+        cmd: 'manage',
+        data: {
+          label: this.label,
+          cmd: {
+            type: 'startDragging'
+          }
+        }
+      }
+    })
+  }
+}
+
+/**
+ * Create new webview windows and get a handle to existing ones.
+ * @example
+ * ```typescript
+ * // loading embedded asset:
+ * const webview = new WebviewWindow('theUniqueLabel', {
+ *   url: 'path/to/page.html'
+ * })
+ * // alternatively, load a remote URL:
+ * const webview = new WebviewWindow('theUniqueLabel', {
+ *   url: 'https://github.com/tauri-apps/tauri'
+ * })
+ *
+ * webview.once('tauri://created', function () {
+ *  // webview window successfully created
+ * })
+ * webview.once('tauri://error', function (e) {
+ *  // an error happened creating the webview window
+ * })
+ *
+ * // emit an event to the backend
+ * await webview.emit("some event", "data")
+ * // listen to an event from the backend
+ * const unlisten = await webview.listen("event name", e => {})
+ * unlisten()
+ * ```
+ */
+class WebviewWindow extends WindowManager {
+  constructor(label: string, options: WindowOptions = {}) {
+    super(label)
+    invokeTauriCommand({
+      __tauriModule: 'Window',
+      message: {
+        cmd: 'createWebview',
+        data: {
+          options: {
+            label,
+            ...options
+          }
+        }
       }
     })
+      .then(async () => this.emit('tauri://created'))
+      .catch(async (e) => this.emit('tauri://error', e))
+  }
+
+  /**
+   * Gets the WebviewWindow handle for the webview associated with the given label.
+   *
+   * @param label The webview window label.
+   * @returns The handle to communicate with the webview or null if the webview doesn't exist.
+   */
+  static getByLabel(label: string): WebviewWindowHandle | null {
```

Comment:
Shouldn't this and `getCurrent()` return `WebviewWindow` or `WindowManager` ? or are we still don't want to let a window manage another ?

Reply:
you're right